tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1516,8 +1516,26 @@ class PlaceholderExpansionScanner {
       bool walkToExprPre(Expr *E) override {
         auto SR = E->getSourceRange();
         if (SR.isValid() && SM.rangeContainsTokenLoc(SR, TargetLoc)) {
-          if (!checkCallExpr(E) && !EnclosingCallAndArg.first)
+          if (auto closure = dyn_cast<ClosureExpr>(E)) {
+            if (closure->hasSingleExpressionBody()) {
+              // Treat a single-expression body like a brace statement and reset
```

Comment:
The handling of closure expressions feels a bit complicated, but I haven't found a way to simplify it.  It would be easier if single-expression closures contained an implicit brace statement (like single-expression functions do), but because it's a pure expression it needs to get handled somewhere.  If you have any ideas, let me know!

Reply:
I couldn't come up with better idea 😓